### PR TITLE
Use spf13/cobra for docker import

### PR DESCRIPTION
**Pull request by vdemeester**
Moves import command to `api/client/image/import.go` and use cobra  🐍.

/cc @dnephin @thaJeztah @LK4D4 @cpuguy83 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

**Comment by vdemeester** [comment]
Hum… there is a weird error on `DockerSuite.TestEventsImageImport`... looking into it

File: api/client/image/import.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func NewImportCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts importOptions
+
+	cmd := &cobra.Command{
+		Use:   "import [OPTIONS] file|URL|- [REPOSITORY[:TAG]]",
+		Short: "Import the contents from a tarball to create a filesystem image",
+		Args:  cli.RequiresMinArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.source = args[0]
+			opts.reference = args[1]
```
Comment (dnephin) [inline api/client/image/import.go]: min args is set to 1, but it looks like this assumes 2.

I think this needs `if len(args) > 1`

Comment (vdemeester) [inline api/client/image/import.go]: Yeah definitely.

**Comment by vdemeester** [comment]
Hum, so `cobra` is eating `-` arguments (without saying anything which is worrying me a bit) and thus `docker import -` is failing with "… requires a last 1 argument(s)".

/cc @dnephin

**Comment by dnephin** [comment]
hmm, I'll look into it later today. Might be a bug in https://github.com/dnephin/cobra/blob/fork/command.go#L355-L393

**Comment by vdemeester** [comment]
@dnephin yeah I was looking into that. Note that it's just the "args" passed on the `Args` function of the command. The "args" passed to the `Run` are ok ; I'll update the PR with a small temporary fix

File: api/client/image/import.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+// NewImportCommand creates a new `docker import` command
+func NewImportCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts importOptions
+
+	cmd := &cobra.Command{
+		Use:   "import [OPTIONS] file|URL|- [REPOSITORY[:TAG]]",
+		Short: "Import the contents from a tarball to create a filesystem image",
+		// Args:  cli.RequiresMinArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// FIXME The Args function won't see any argument that start with a "-" and
```
Comment (vdemeester) [inline api/client/image/import.go]: Added a `FIXME` :angel:

**Comment by dnephin** [comment]
@vdemeester I pushed a fix for this. If you update the vendor to 3bfe23ade4a4657b00fb065297461ee282dfb822 it should work correctly.

The fix is here: https://github.com/spf13/cobra/pull/284/commits/c76d2faa30cf97bd5c88152b5c261068e4c010d5

**Comment by vdemeester** [comment]
@dnephin are you sure on the hash, it removes the `SetFlagErrorFunc` function and such :sweat:

**Comment by dnephin** [comment]
hmm, it shouldn't remove anything.

Try `75205f23b3ea70dc7ae5e900d074e010c23c37e9`. I rebased it on `spf13/master` again.

This should be right: https://github.com/dnephin/cobra/commits/fork

**Comment by thaJeztah** [comment]
ping @vdemeester this needs a rebase now :cry:

**Comment by dnephin** [comment]
LGTM

File: hack/vendor.sh [hunk 3]
```diff
@@ -139,7 +139,7 @@ clone git github.com/docker/docker-credential-helpers v0.3.0
 clone git github.com/docker/containerd 57b7c3da915ebe943bd304c00890959b191e5264
 
 # cli
-clone git github.com/spf13/cobra acf60156558542e78c6f3695f74b0f871614ff55 https://github.com/dnephin/cobra.git
+clone git github.com/spf13/cobra 75205f23b3ea70dc7ae5e900d074e010c23c37e9 https://github.com/dnephin/cobra.git
```
Comment (thaJeztah) [inline hack/vendor.sh]: should we put this fork under the "docker" organization if we're unsure our changes get accepted before 1.12?

**Comment by thaJeztah** [comment]
LGTM, thanks!

File: hack/vendor.sh (hunk 3 above)
Comment (vdemeester) [inline hack/vendor.sh]: ping @dnephin I would be fine for it :angel:

Comment (dnephin) [inline hack/vendor.sh]: That would be fine with me. There seems to be very little activity on my PRs, so I think it's unlikely they will be merged before 1.12.

**Pull request merged**